src/hpc/autoscale/cost/azurecost.py

```diff
@@ -0,0 +1,233 @@
+                        region = d['region']
+                        core_count = d['core_count']
+                        hours = d['hours']
+                        spot = False
+                        if d['priority'] == 'Spot':
+                            spot = True
+                        if self.do_meter_lookup(sku_name=vm_size,region=region,spot=spot):
+                            pass
+                        rate_data = self.get_retail_rate(armskuname=vm_size, armregionname=region, spot=spot)
+                        rate = rate_data['retailPrice']
+                        cost = (hours/core_count) * rate
+                        array_fmt = az_fmt_t(start=start, end=end, nodearray=array, core_hours=hours, cost=cost,
+                                             sku_name=vm_size,region=region,spot=spot,meterid=rate_data['meterId'],
+                                             metercat=rate_data['serviceName'], rate=rate_data['retailPrice'],
+                                             meter=rate_data['meterName'], currency=rate_data['currencyCode'])
+                        row = []
+                        for f in az_fmt_t._fields:
+                            row.append(array_fmt._asdict()[f])
+                        writer.writerow(row)
+
+        usage = self.get_usage(start, end, 'hourly')
+        _fmt = self.get_nodearray_hourly_format()
+        if _fmt.__name__ == 'az_array_hourly_retail_t':
+            return _process_hourly_with_retail(_fmt, fout, usage)
+
+    def get_retail_rate(self, armskuname: str, armregionname: str, spot: bool):
+
+        params = {}
+        filters = f"armRegionName eq '{armregionname}' and armSkuName eq '{armskuname}' and serviceName eq 'Virtual Machines'"
+        params['$filter'] = filters
+
+        res = self.retail_session.get(self.retail_url, params=params)
+        if res.status_code != 200:
+            log.error(f"{res.json()}")
+            raise res.raise_for_status()
+        data = res.json()
+
+        for e in data['Items']:
+            if e['type'] != 'Consumption':
+                continue
+
+            if e['productName'].__contains__("Windows"):
+                continue
+
+            if e['meterName'].__contains__("Low Priority"):
+                continue
+
+            if e['meterName'].__contains__("Spot"):
+                if spot:
+                    return e
+                continue
+            return e
+
+    def get_info_from_retail(self, meterId: str):
+
+        sku = 'armSkuName'
+        region = 'armRegionName'
+        filters = f"meterId eq '{meterId}'"
+        params = {}
+        params['$filter'] = filters
+
+        res = self.retail_session.get(self.retail_url, params=params)
+        if res.status_code != 200:
+            log.error(f"{res.json()}")
+            raise res.raise_for_status()
+
+        data = res.json()
+        sku_list = []
+        for e in data['Items']:
+            if e[sku] and e[region]:
```

Comment:
It definitely _should_ have sku and region as keys, but if it does not it will fail with a strange KeyError.
The pythonic way to do this check is like this:
if e.get(sku) and e.get(region):

Reply:
Fixed